### PR TITLE
Make Mark All As Read optional

### PR DESCRIPTION
**Pull request by karlhorky**
I just noticed the new "Mark all as read" link that was added in this PR by @tjx666:

- https://github.com/tanmayrajani/notifications-preview-github/pull/152

It seems like a nice feature! Maybe even good enough to be default for most users 👍 

But for me (and maybe also others), it makes me nervous that I will accidentally click on it and mark everything as read. I often keep notifications as a kind of ad-hoc todo list for myself.

Also, it takes up extra vertical space, which I would like to avoid:

Screenshot 2024-08-17 at 18 02 14

So I added an option (default: link enabled), which allows users to disable the link if they like.

cc @fregante what do you think?

### CODE REVIEW

**Review by fregante — APPROVED** [review]
@tjx666 thoughts?

File: source/github-notifications-preview.jsx [hunk 1]
```diff
@@ -163,12 +164,16 @@ async function openDropdown({currentTarget: indicator}) {
 					>
 						<path d='M2.5 1.75v11.5c0 .138.112.25.25.25h3.17a.75.75 0 0 1 0 1.5H2.75A1.75 1.75 0 0 1 1 13.25V1.75C1 .784 1.784 0 2.75 0h8.5C12.216 0 13 .784 13 1.75v7.736a.75.75 0 0 1-1.5 0V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13.274 9.537v-.001l-4.557 4.45a.75.75 0 0 1-1.055-.008l-1.943-1.95a.75.75 0 0 1 1.062-1.058l1.419 1.425 4.026-3.932a.75.75 0 1 1 1.048 1.074ZM4.75 4h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1 0-1.5ZM4 7.75A.75.75 0 0 1 4.75 7h2a.75.75 0 0 1 0 1.5h-2A.75.75 0 0 1 4 7.75Z'/>
 					</svg>
-                    &nbsp;&nbsp;
+		&nbsp;&nbsp;
```
Comment (karlhorky) [inline source/github-notifications-preview.jsx:167]:
```suggestion
                    &nbsp;&nbsp;
```

File: source/github-notifications-preview.jsx [hunk 2]
```diff
@@ -206,7 +211,7 @@ async function openDropdown({currentTarget: indicator}) {
 
 			// All notifications are read, remove the header
 			if ($$('.js-notifications-list-item, .js-notifications-group').length === 0) {
-				dropdownHeader.remove();
+				if (options.markAllAsRead) dropdownHeader.remove();
```
Comment (karlhorky) [inline source/github-notifications-preview.jsx:214]:
```suggestion
				if (options.markAllAsRead) {
					dropdownHeader.remove();
				}
```

File: source/github-notifications-preview.jsx [hunk 3]
```diff
@@ -206,7 +211,9 @@ async function openDropdown({currentTarget: indicator}) {
 
 			// All notifications are read, remove the header
 			if ($$('.js-notifications-list-item, .js-notifications-group').length === 0) {
-				dropdownHeader.remove();
+				if (options.markAllAsRead) {
+					dropdownHeader.remove();
+				}
```
Comment (karlhorky) [inline source/github-notifications-preview.jsx:216]:
```suggestion
				}

```

**Comment by karlhorky** [comment]
Apart from the approval by @tjx666, anything else I can do in this PR to make it ready?

**Comment by tjx666** [comment]
fine.

**Comment by karlhorky** [comment]
Thanks for the review, merge and publish of [`notifications-preview-github@24.8.19`](https://github.com/fregante/notifications-preview-github/releases/tag/24.8.19) 🙌

**Pull request merged**